tests/kernel/common/src/atomic.c

```diff
@@ -267,6 +279,49 @@ void test_atomic(void)
 			"Failed to clear a single bit in an array of atomic variables");
 	}
 }
+
+void atomic_handler(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	for (int i = 0; i < TEST_CYCLE; i++) {
+		atomic_inc(&total_atomic);
+	}
+}
+
+/**
+ * @brief Verify atomic operation with threads
+ *
+ * @details Check if the atomic operation go correct when multiple
+ * threads access the same atomic value.
+ *
+ * @ingroup kernel_common_tests
+ */
+void test_threads_access_atomic(void)
+{
+	k_tid_t tid[THREADS_NUM];
+
+	/* enable time slice 1ms at priority 10 */
+	k_sched_time_slice_set(1, K_PRIO_PREEMPT(10));
```

Comment:
why is time slicing relevant here?

Reply:
Although CONFIG_TIMESLICING is by default, the default CONFIG_TIMESLICE_SIZE is 0. A time slice size of zero means "no limit" (i.e. an infinitely large time slice).
So I use function k_sched_time_slice_set to set the time slice at a very small value, it make the preempt thread which have equal thread priority can be scheduling by timeslice.

Reply:
@nashif The previous description is not clear. The testcase description is update to "This is a testcase for testing if two preempt child threads with equal priority and time slice 1ms can atomiclly access the same atomic value." The 2 preempt child threads with equal priority can not be preempted with default timeslice 0. That's why I need set time slice to 1ms.
Hope this clarifies.